### ISSUE TITLE
General Retrospective for March 2022 Releases

### ISSUE THREAD

**Issue by adamfarley** [adoptium/adoptium#128]
**Summary**

A retrospective for all efforts surrounding the titular releases. 

All community members are welcome to contribute to the agenda via comments below.

This will be a virtual meeting after the release, with at least a week of notice in the #release Slack channel.

On the day of the meeting the agenda items will be iterated over, with a list of actions added at in a comment at/near the end.

Invited: Everyone.

**Time, Date, and URL**

Time: 1:30PM GMT+1 (10:30am Ontario)
Date: 13th of April

URL: [https://bluejeans.com/917304424](https://www.google.com/url?q=https://bluejeans.com/917304424&sa=D&source=calendar&ust=1649842322077530&usg=AOvVaw39_Rbj5YqwL9c5Sk4qlSWX)

**Details**

Retrospective Owner Tasks (in order):

- [x] Post this issue's URL on the \#Release Slack channel around the start of the new release.
- [x] Copy actions from the previous retrospective into this issue, while ignoring actions that are ticked or have issue links (i.e. clearly-completed actions).
- [ ] Wait until 90% of all builds have been released, with no clear signs of additional respins.
- [ ] Announce the retrospective Slack call's date + time on \#Release at least one full week in advance, and send out meeting invites.
- [ ] Host the slack call for the retrospective, including:
  - [ ] Iterating over the actions from the previous retrospective issue, ticking off completed items.
  - [ ] Iterate over the agenda, ensuring everything gets debated.
  - [ ] Create a clear list of actions at the end of the retrospective, including volunteer names.
- [ ] Create a new retrospective issue for the next release.
- [ ] Set yourself a calendar reminder so that you remember to commence step 1 (in the new issue) just before the next release.
- [ ] Close this issue.

**Comment by adamfarley** [comment]
This item was carried over from the [last retrospective](https://github.com/adoptium/adoptium/issues/104#issuecomment-1036065992):

\[sxa\]: Can we do anything with the linuxNew packaging job to make it more obvious when it is blocked on a machine not being available so remedial action can be taken? (Note: Currently looking at the BlueOcean view seems to be the best way, or possibly pipeline steps)

**Comment by smlambert** [comment]
Notes for retrospective / suggestions for improvements (in no particular order):
- archive the old release doc (which contains lots of openj9 details that are no longer used at our project)
- create a smaller/leaner release doc divided into sections that align with the checklist, link to each specific section from the checklist (reducing random searching for the applicable information)
- release tool could be improved with dropdown choice for which platform(s) to release and when chosen, the right regex filled out for you (versus needing to fill in / update the ARTIFACTS_TO_COPY field, also add **/*testimage* as the example for ARTIFACTS_TO_SKIP
- suggest a better format for AQA triage and final assessment (to reduce effort to find the final assessment) - link to a comment in the issue from the description, comment in issue ultimately says 👍 or 👎 after triage completes.
- mirror job work when a new java version is available (we shifted to jdk18u, but did not have all of the tags needed... how to not hit this problem again in the future).  We lost the opportunity to have a build ready (and AQA testing complete) in advance of the GA tag.

**Comment by smlambert** [comment]
Clarify in documentation, if Homebrew cask would/should be updated with STS release information? (its currently the latest jdk17 LTS release)

For installers, for new versions, presume one needs to add entirely new directories in the intaller repo, this needs to be more clear also (with link to a similar PR which will be created this March release for the first time?)

**Comment by smlambert** [comment]
Update release_notes section of https://github.com/gdams/openjdk-build/blob/releasing/RELEASING.md#steps-for-every-version, since the new website does not seem to have release_notes documentation.

**Comment by sxa** [comment]
Ensure we have documentation on the processes for creating repositories for new major java versions

**Comment by smlambert** [comment]
Remove outdated doc at https://github.com/adoptium/installer#releasing-linux-installer-packages and add reference to the new Jenkins job (in the releasing doc / checklist and in installer doc) - https://ci.adoptopenjdk.net/job/adoptium-packages-linux-pipeline/

**Comment by smlambert** [comment]
Release notes blocked by no release notes page in website-v2 (https://github.com/adoptium/website-v2/issues/103)

**Comment by smlambert** [comment]
Update release-status [template](https://github.com/adoptium/adoptium/blob/main/.github/ISSUE_TEMPLATE/release-status.md) to remove hyperlinks for builds (not accessible to general public anyway).

**Comment by smlambert** [comment]
Other lessons as a trainee:
For the homebrew cask PR, I did not initially grab the right sha256 value (grabbed the one for the tarball instead of the pkg file).
aarch64 MacOS pkg did not push up properly to releases repo when I published it (I believe related to a recently fixed issue refactor_release_tool needs to use jdk8), and it had to be re-pushed up after this [change](https://adoptium.slack.com/archives/C09NW3L2J/p1649331418961299?thread_ts=1648734385.820889&cid=C09NW3L2J)

**Comment by smlambert** [comment]
https://github.com/adoptium/adoptium/issues/131#issuecomment-1089373975

**Comment by sxa** [comment]
**ACTIONS:**
- Improvements to how to use the release tool (Maybe have platform dropdowns?): https://github.com/adoptium/temurin-build/issues/2847
- Make it easier to determine if linuxNew installer job is stalled on a particular machine/platform: https://github.com/adoptium/installer/issues/453
- Shelley to update docs relating to homebrew on how STS releases are published (default stream info etc.)
- Define policy relating to publishing release notes on the new website (if we choose to do so)
- Shelley to 'squash' (turn into twisties) the OpenJ9 information in the releasing guide now that we are no longer shipping Openj9 (@AdamBrousseau do people doing any of the openj9 releases use the adoptium releasing guide?)

**Comment by AdamBrousseau** [comment]
>do people doing any of the openj9 releases use the adoptium releasing guide?

- When we started doing Semeru releases I "forked" your release checklist. I think yours has changed since and I haven't updated ours to follow those changes. I don't follow the [RELEASING.md](https://github.com/adoptium/temurin-build/blob/master/RELEASING.md) and I would be surprised if anyone else does.

**Comment by karianna** [comment]
For those watching, linuxNew has been renamed to linux
